### ISSUE TITLE
CP1251 and other CP encodings support in media tags

### ISSUE THREAD

**Issue by Yanpas** [clementine-player/Clementine#4788]
A lot of music with cyrillic tags ancoded in CP1251. And I don't know any player in Linux that supports it. Moreover there is no support of cp1251 in windows, where it is default encoding

**Comment by user-42** [comment]
Use the Unicode, Luke.

**Comment by Yanpas** [comment]
iTunes supports both encodings. Maybe do not listen non-latin music?:D

**Comment by Tatsh** [comment]
That is not true @Yanpas. I have extreme doubts that iTunes supports both encodings, as they program that in Objective-C (yes even on Windows), where the encoding of strings is *always* Unicode.

If you are on Windows, It is likely your system locale is set to a Cyrillic-writing language (like Russian) which means any non-Unicode text will be auto-converted from the most common text codec (CP1251) to Unicode (specifically, UTF-32) and you will not see it. iTunes does not see it either as the conversion is transparent to all software. You should try setting your Windows machine locale to English, rebooting, and watch how iTunes screws up badly at showing your tags (along with anything else that assumes they are just regular Unicode or ASCII).

I agree with @user-42. Convert your tags to Unicode. There is nothing in CP1251 that cannot be represented in Unicode fully. There are tools to do this. Even on Linux you could combine iconv with id3ted on the command line. This shows going from UTF-8 to CP1251 and back to UTF-8:

```
# Пётр Ильич Чайкoвский -> Piotr Il'ich Chaikovskii (via unidecode)
$ id3ted -l some.mp3 | fgrep TPE1
TPE1: Пётр Ильич Чайкoвский
$ id3ted -l some.mp3 | fgrep TPE1 | iconv --from-code=UTF-8 --to-code=CP1251
TPE1: <garbage>
$ id3ted -l some.mp3 | fgrep TPE1 | iconv --from-code=UTF-8 --to-code=CP1251 | iconv --from-code=CP1251 --to-code=UTF-8
TPE1: Пётр Ильич Чайкoвский
```

So if `id3ted` gives you garbage for TPE1, TIT2, etc, then you can convert with iconv. You can even set the tag on the fly. This shows setting the artist:

```
id3ted -a "$(id3ted -l some.mp3 | fgrep TPE1 | sed -e 's/^TPE1: //' | iconv --from-code=CP1251 --to-code=UTF-8)" some.mp3
```

**Comment by Yanpas** [comment]
hm, so why windows version of Clementine doesn't behave like iTunes (doesn't automatically convert cp1251 to utf8 in background) ? There is also DeadBeef player which automatically recognizes cp1251 on Linux (i haven't found anywhere in settings mention of cp1251).

Anyway mid3iconv -eCP1251 *mp3 helps me
(to install it - install python-mutagen)

**Comment by nikolayo** [comment]
My bit to this : I use the Linux version of Clementine 1.2 on Ubuntu 14.04. My locale is en_US.UTF-8. Clementine plays fine tracks with titles in Cyrillic. Titles are displayed correctly in the file browser but come up garbled in the library browser and in the playlist tabs.

**Comment by TheBestDenis** [comment]
For the information of dear audiophiles and IT specialists. The CP1251 encoding has been used for more than a dozen years. If you respect yourself, then respect others as well. Support for this encoding is necessary. Are you programmers? Or maybe losers?

**Comment by fitojb** [comment]
@TheBestDenis LMAO.

**Comment by 0x131315** [comment]
Lol, still it is not fixed? omg

**Comment by alien190** [comment]
Oh, I have the same trouble.
